akka-testkit/src/main/scala/akka/testkit/ExplicitlyTriggeredScheduler.scala

```diff
@@ -69,19 +69,19 @@ class ExplicitlyTriggeredScheduler(@unused config: Config, log: LoggingAdapter,
     currentTime.set(newTime)
   }
 
-  private def scheduledTasks(runTo: Long): Seq[Item] =
-    scheduled.keySet().asScala.filter(_.time <= runTo).toList.sortBy(_.time)
+  private def scheduledTasks(runTo: Long): Seq[(Item, Long)] =
+    scheduled.entrySet().asScala.map(s => (s.getKey, s.getValue)).toSeq.filter { case (_, v) => v <= runTo }.sortBy(_._2)
 
   @tailrec
   private[testkit] final def executeTasks(runTo: Long): Unit = {
     scheduledTasks(runTo).headOption match {
-      case Some(task) =>
-        currentTime.set(task.time)
+      case Some((task, time)) =>
+        currentTime.set(time)
         val runResult = Try(task.runnable.run())
         scheduled.remove(task)
 
         if (runResult.isSuccess)
-          task.interval.foreach(i => scheduled.put(task.copy(time = task.time + i.toMillis), ()))
```

Comment:
Before we were updating the task with new execution time. Because of this, the Cancellable was left with stale reference (this line: https://github.com/akka/akka/pull/28614/files#diff-7a1a738a6673f4092734cfa193957d7dR109), and hence the task could not be canceled (=removed from the scheduled map).